packages/ensnode-sdk/src/tokenscope/name-token.ts

```diff
@@ -0,0 +1,129 @@
+import { type Address, zeroAddress } from "viem";
+
+import type { AccountId } from "../shared";
+import {
+  type DomainAssetId,
+  type NFTMintStatus,
+  type SerializedDomainAssetId,
+  serializeDomainAssetId,
+} from "./assets";
+
+/**
+ * An enum representing the possible Name Token Ownership types.
+ */
+export const NameTokenOwnershipTypes = {
+  Proxy: "proxy",
+  Effective: "effective",
+  Burned: "burned",
+} as const;
+
+export type NameTokenOwnershipType =
+  (typeof NameTokenOwnershipTypes)[keyof typeof NameTokenOwnershipTypes];
+
+export interface NameTokenOwnershipProxy {
+  ownershipType: typeof NameTokenOwnershipTypes.Proxy;
+
+  /**
+   * Owner
+   *
+   * Guarantees:
+   * - `owner.address` is not the zero address.
+   */
+  owner: AccountId;
+}
+
+export interface NameTokenOwnershipEffective {
+  ownershipType: typeof NameTokenOwnershipTypes.Effective;
+
+  /**
+   * Owner
+   *
+   * Guarantees:
+   * - `owner.address` is not the zero address.
+   */
+  owner: AccountId;
+}
+
+export interface NameTokenOwnershipBurned {
+  ownershipType: typeof NameTokenOwnershipTypes.Burned;
+
+  /**
+   * Owner
+   *
+   * Guarantees:
+   * - `owner.address` is the zero address.
+   */
+  owner: AccountId;
+}
+
+export type NameTokenOwnership =
+  | NameTokenOwnershipProxy
+  | NameTokenOwnershipEffective
+  | NameTokenOwnershipBurned;
+
+export interface NameToken {
+  /**
+   * Domain Asset
+   *
+   * References NFT that tokenizes the ownership of a domain.
+   */
+  domainAsset: DomainAssetId;
```

Comment:
```suggestion
  token: AssetId;
```

Goals:
1. Won't it be redundant to keep repeating the same `domainId` value in each `NameToken` that's returned for the same name? Therefore suggesting to change this field from a `DomainAssetId` to just an `AssetId`. I also note how `RegisteredNameTokens` already includes both a `domainId` and `name` field already so seems really no need to duplicate `domainId` further for each `NameToken` in the response.
2. I imagine code that uses this object will read nicer if we rename this field from `domainAsset` to `token`.